Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingPrivacyViewModel.kt

```diff
@@ -1,16 +1,28 @@
 package de.rki.coronawarnapp.ui.onboarding
 
+import androidx.lifecycle.asLiveData
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.eol.AppEol
+import de.rki.coronawarnapp.util.coroutine.DefaultDispatcherProvider
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
 
-class OnboardingPrivacyViewModel @AssistedInject constructor() : CWAViewModel() {
+class OnboardingPrivacyViewModel @AssistedInject constructor(
+    eol: AppEol,
+    dispatcherProvider: DispatcherProvider = DefaultDispatcherProvider(),
+) : CWAViewModel() {
     val routeToScreen: SingleLiveEvent<OnboardingNavigationEvents> = SingleLiveEvent()
+    val isEol = eol.isEol.asLiveData(context = dispatcherProvider.Default)
```

Comment:
You are exposing `isEol` from the view model to the fragment which keeps track of the is flag and then when a button is clicked you pass the flag back to the view model. why don’t you just check in the viewmodel upon button click ?
```kotlin
fun onNextButtonClick() = launch{
 routeToScreen.postValue( 
    if(eol.isEol.first()) NavigateToMainActivity else NavigateToOnboardingTracing
 )
}
```

Reply:
Yes, true. I'll update it.